clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractPartitionAssignor.java

```diff
@@ -42,8 +42,24 @@
      * @param subscriptions Map from the memberId to their respective topic subscription
      * @return Map from each member to the list of partitions assigned to them.
      */
-    public abstract Map<String, List<TopicPartition>> assign(Map<String, Integer> partitionsPerTopic,
-                                                             Map<String, Subscription> subscriptions);
+    public Map<String, List<TopicPartition>> assign(Map<String, Integer> partitionsPerTopic,
```

Comment:
It might be safer to leave this abstract. Otherwise we should choose a better sentinel for the generation. Perhaps we should change the API below to use `Optional<Integer>` instead?

Reply:
👍